standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/MetaStoreDirectSql.java

```diff
@@ -1248,30 +1253,43 @@ protected boolean shouldStop() {
     }
 
     private static enum FilterType {
-      Integral,
-      String,
-      Date,
-
-      Invalid;
-
-      static FilterType fromType(String colTypeStr) {
-        if (colTypeStr.equals(ColumnType.STRING_TYPE_NAME)) {
-          return FilterType.String;
-        } else if (colTypeStr.equals(ColumnType.DATE_TYPE_NAME)) {
-          return FilterType.Date;
-        } else if (ColumnType.IntegralTypes.contains(colTypeStr)) {
-          return FilterType.Integral;
+      Integral("tinyint,smallint,int,bigint", "java.lang.Long"),
+      String("string", "java.lang.String"),
+      Date("date", "java.sql.Date"),
+      Timestamp("timestamp", "java.sql.Timestamp"),
+
+      Invalid("Invalid", "null");
+
+      private final List<String> type;
+      private final String value;
```

Comment:
Instead of String parameters could you please define the proper types of the arguments:
```
Integral(ImmutableSet.of(INT_TYPE_NAME, SMALLINT_TYPE_NAME, ...), Long.class),
String(ImmutableSet.of(STRING_TYPE_NAME), String.class),
...
Invalid(Collections.emptySet(), Void.class);

private final Set<String> colTypes;
private final Class<?> clazz;
```

Reply:
ack.